File: clients/dart/bots/random_ai.dart
import 'dart:math';
import '../play_state.dart';
import 'bot_ai.dart';

class RandomAI implements BotAi {
  int _column = 0;
  final Random _random = Random();

  RandomAI();

  @override
  int play(PlayState playState) {
    _column = _random.nextInt(7);
    return _column;
  }

  @override
  String getName() {
    return "RandomBot";
  }
}

File: clients/typescript/src/bots/randomAI.ts
import {BotAI} from './botAI';
import {PlayState} from './data';

export class RandomAI implements BotAI {
    private name: string = 'TypeScriptTestBot';

    play(currentGameState: PlayState): number {
        console.log("Your Coin ID is: " + currentGameState.coin_id);
        console.log(currentGameState.board)
        const randomColumn: number = Math.floor(Math.random() * 7);
        return randomColumn;
    }

    getName(): string {
        return this.name;
    }
}
